<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Mercedes Guerrero, Alba Serrano and Alex Sanchez" />

<meta name="date" content="2019-09-09" />

<title>Examples of microarray analysis using BasicP4microArrays functions</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Examples of microarray analysis using <code>BasicP4microArrays</code> functions</h1>
<h4 class="author">Mercedes Guerrero, Alba Serrano and Alex Sanchez</h4>
<h4 class="date">2019-09-09</h4>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The <code>BasicP4microArrays</code> package has been developed to simplify the analysis of microarray data, especially in those cases where there may be many comparisons in a study or in cases where different subsets of the same dataset have to be analyzed in similar or distinct ways.</p>
<p>In a simplified manner such a workflow consists of the following steps:</p>
<ol style="list-style-type: decimal">
<li>Read the raw data</li>
<li>[Optional]Check its quality</li>
<li>Normalize the data</li>
<li>Provide visualizations of normalized data
<ol style="list-style-type: decimal">
<li>To check its quality</li>
<li>To detect potential batch effects</li>
</ol></li>
<li>Filter the normalized data</li>
<li>Select differentially expressed genes using a linear model.</li>
<li>Provide visualizations of
<ol style="list-style-type: decimal">
<li>differentially expressed genes (“volcanos”)</li>
<li>common and distinct genes selected between comparisons (“venn diagrams”)</li>
<li>groups of genes and samples “heatmaps”)</li>
</ol></li>
<li>Do a biological significance analysis based on the Gene Ontology</li>
<li>Generate a report/summary of analysis results.</li>
</ol>
<p>The <code>BasicP4microArrays</code> package implement a few “main” functions which together support a typical microarray analysis workflow. That is, ignoring the parameters of these functions the previous workflow could be implemented with several calls such as:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(BasicP4microArrays)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="cf">if</span>(<span class="op">!</span>(<span class="kw">require</span>(hgu133a2))) BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;hgu133a2.db&quot;</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">rawData &lt;-<span class="st">        </span>BasicP4microArrays<span class="op">::</span><span class="kw">readOrLoad.RawData</span>(...)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">anotacions &lt;-<span class="st">     </span>BasicP4microArrays<span class="op">::</span><span class="kw">createOrLoadAnnotations</span>(...)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">eset_norm &lt;-<span class="st">      </span>BasicP4microArrays<span class="op">::</span><span class="kw">normalization</span>(<span class="dt">my.data =</span> rawData, ...)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6">                  BasicP4microArrays<span class="op">::</span><span class="kw">normplots2File</span>(<span class="dt">my.data =</span> eset_norm, ...)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7">exprs.filtered &lt;-<span class="st"> </span>BasicP4microArrays<span class="op">::</span><span class="kw">filterData</span>(<span class="dt">expres =</span> <span class="kw">exprs</span>(eset_norm), ...)</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">fitMain &lt;-<span class="st">        </span>BasicP4microArrays<span class="op">::</span><span class="kw">lmAnalysis</span>(<span class="dt">exprs.filtered =</span> exprs.filtered, ...)</a>
<a class="sourceLine" id="cb1-9" data-line-number="9">genesAnnotated &lt;-<span class="st"> </span>BasicP4microArrays<span class="op">::</span><span class="kw">GeneAnnotation</span>(<span class="dt">egIDs =</span> genes2annotate, ...)</a>
<a class="sourceLine" id="cb1-10" data-line-number="10">clust &lt;-<span class="st">          </span>BasicP4microArrays<span class="op">::</span><span class="kw">clusterAnalysis</span>(<span class="dt">expres =</span> exprs.filtered, ...)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11">GOResult &lt;-<span class="st">       </span>BasicP4microArrays<span class="op">::</span><span class="kw">GOAnalysis</span>(<span class="dt">fitMain =</span> fitMain, ...)</a></code></pre></div>
<p>The tricky thing to use these functions is, indeed, to set the value of all parameters needed and this is illustrated in the examples below.</p>
<p>Besides this, it is possible to do more complex analyses using iterator functions that can call repeteadly some of the analysis functions so that they can be applied to different subsets of the data or with distinct design or contrast matrices.</p>
</div>
<div id="use-case-1-basic-microarray-analysis-using-3ivt-microarrays" class="section level1">
<h1>Use case 1: Basic microarray analysis using 3’IVT microarrays</h1>
<p>First of all, we will declare the directories we will use:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">workingDir &lt;-<span class="st"> </span><span class="kw">getwd</span>()</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">dataDir    &lt;-<span class="st"> </span><span class="kw">file.path</span>(workingDir, <span class="st">&quot;datafiles-ex1&quot;</span>)</a></code></pre></div>
<div id="the-function-readorload.rawdata" class="section level3">
<h3>The function <code>readOrLoad.RawData</code></h3>
<p>This function reads data from cel files or loads previously saved data.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">readCELS &lt;-<span class="st"> </span><span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">my.targets      &lt;-<span class="st"> </span><span class="kw">file.path</span>(dataDir, <span class="st">&quot;targets.txt&quot;</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">targets         &lt;-<span class="st"> </span><span class="kw">read.table</span>(my.targets,</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">                              <span class="dt">head =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">                              <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">                              <span class="dt">row.names =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">my.fileNames    &lt;-<span class="st"> </span><span class="kw">paste</span>(dataDir,</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">                         <span class="kw">rownames</span>(targets),</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">                         <span class="dt">sep =</span> <span class="st">&quot;/&quot;</span>)</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">rawDataFileName &lt;-<span class="st"> &quot;rawData.Rda&quot;</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11">my.outputDir    &lt;-<span class="st"> &quot;.&quot;</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12">isExonStudy     &lt;-<span class="st"> </span><span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13">orgPackage      &lt;-<span class="st"> &quot;org.Hs.eg&quot;</span></a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">readCELS</a></code></pre></div>
<pre><code>[1] TRUE</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">my.targets</a></code></pre></div>
<pre><code>[1] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/targets.txt&quot;</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">targets</a></code></pre></div>
<pre><code>               Sample             Ids ShortName Group Apocrine.grade
GSM26878.CEL GSM26878 PF14 EnPnT2N1G2      PF14     A              3
GSM26883.CEL GSM26883 PF19 EnPuT4N0Gu      PF19     A              2
GSM26887.CEL GSM26887 PF23 EnPnT2N0G2      PF23     A              3
GSM26903.CEL GSM26903 PF39 EnPuT4N0Gu      PF39     A              3
GSM26910.CEL GSM26910 PF46 EnPnT4N1G3      PF46     A              3
GSM26888.CEL GSM26888 PF24 EnPnTiN0G3      PF24     B              2
GSM26889.CEL GSM26889 PF25 EnPnT3N2G2      PF25     B              1
GSM26892.CEL GSM26892 PF28 EnPnT2N1G3      PF28     B              2
GSM26898.CEL GSM26898 PF34 EnPnT3N1G3      PF34     B              1
GSM26906.CEL GSM26906 PF42 EnPnT2N2G3      PF42     B             na
GSM26879.CEL GSM26879 PF15 EpPnTiN1G3      PF15     L              2
GSM26896.CEL GSM26896 PF32 EpPnT3N1G2      PF32     L              2
GSM26897.CEL GSM26897 PF33 EpPnTiN0G2      PF33     L              1
GSM26907.CEL GSM26907 PF43 EpPpT2N1G2      PF43     L              1
GSM26911.CEL GSM26911 PF47 EpPpT3N1G3      PF47     L              3
             AR.repeat.length
GSM26878.CEL               18
GSM26883.CEL               18
GSM26887.CEL               18
GSM26903.CEL               17
GSM26910.CEL               19
GSM26888.CEL               NA
GSM26889.CEL               21
GSM26892.CEL               22
GSM26898.CEL               17
GSM26906.CEL               16
GSM26879.CEL               16
GSM26896.CEL               23
GSM26897.CEL               18
GSM26907.CEL               20
GSM26911.CEL               17</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">my.fileNames</a></code></pre></div>
<pre><code> [1] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26878.CEL&quot;
 [2] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26883.CEL&quot;
 [3] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26887.CEL&quot;
 [4] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26903.CEL&quot;
 [5] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26910.CEL&quot;
 [6] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26888.CEL&quot;
 [7] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26889.CEL&quot;
 [8] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26892.CEL&quot;
 [9] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26898.CEL&quot;
[10] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26906.CEL&quot;
[11] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26879.CEL&quot;
[12] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26896.CEL&quot;
[13] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26897.CEL&quot;
[14] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26907.CEL&quot;
[15] &quot;/home/aserrano/GitHub/BasicP4microArrays/vignettes/datafiles-ex1/GSM26911.CEL&quot;</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">rawDataFileName</a></code></pre></div>
<pre><code>[1] &quot;rawData.Rda&quot;</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">my.outputDir</a></code></pre></div>
<pre><code>[1] &quot;.&quot;</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">isExonStudy</a></code></pre></div>
<pre><code>[1] FALSE</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">orgPackage</a></code></pre></div>
<pre><code>[1] &quot;org.Hs.eg&quot;</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">rawData &lt;-<span class="st"> </span>BasicP4microArrays<span class="op">::</span><span class="kw">readOrLoad.RawData</span>(<span class="dt">readCELS =</span> readCELS,</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">                                                  <span class="dt">phenoDat =</span> my.targets,</a>
<a class="sourceLine" id="cb20-3" data-line-number="3">                                                  <span class="dt">fileNames =</span> my.fileNames,</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">                                                  <span class="dt">dataFName =</span> rawDataFileName,</a>
<a class="sourceLine" id="cb20-5" data-line-number="5">                                                  <span class="dt">outputDir =</span> my.outputDir,</a>
<a class="sourceLine" id="cb20-6" data-line-number="6">                                                  <span class="dt">exonSt =</span> isExonStudy)</a></code></pre></div>
</div>
<div id="the-function-createorloadannotations" class="section level3">
<h3>The function <code>createOrLoadAnnotations</code></h3>
</div>
<div id="the-function-normalization" class="section level3">
<h3>The function <code>normalization</code></h3>
</div>
<div id="the-function-normplots2file" class="section level3">
<h3>The function <code>normplots2File</code></h3>
</div>
<div id="the-function-filterdata" class="section level3">
<h3>The function <code>filterData</code></h3>
</div>
<div id="the-function-lmanalysis" class="section level3">
<h3>The function <code>lmAnalysis</code></h3>
</div>
<div id="the-function-geneannotation" class="section level3">
<h3>The function <code>GeneAnnotation</code></h3>
</div>
<div id="the-function-clusteranalysis" class="section level3">
<h3>The function <code>clusterAnalysis</code></h3>
</div>
<div id="the-function-goanalysis" class="section level3">
<h3>The function <code>GOAnalysis</code></h3>
</div>
</div>
<div id="use-case-2-basic-microarray-analysis-using-exon-clariom-s-microarrays" class="section level1">
<h1>Use case 2: Basic microarray analysis using exon (<code>clariom-s</code>) microarrays</h1>
</div>
<div id="use-case-3-a-complex-analysis-1-using-iterators-and-a-single-design-matrix" class="section level1">
<h1>Use case 3: A complex analysis (1) using iterators and a single design matrix</h1>
</div>
<div id="use-case-4-a-complex-analysis2-with-paired-and-unpaired-data" class="section level1">
<h1>Use case 4: A complex analysis(2) with paired and unpaired data</h1>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
